### ISSUE TITLE
children propType does not handle conditional elements

### ISSUE THREAD

**Issue by camjackson** [yearofthedan/layabout#3]
```
const someCondition = false;
const MyThing = () => (
  <FlexRow>
    {someCondition && <span />}
  </FlexRow>
);
```

This will cause a child of `false` to get passed to the `FlexRow`, which is valid React, but it fails the propType of `oneOfType([element, arrayOf(element)])`. I think if you just use `React.PropTypes.node`, it will handle all the edge cases for you RE what is and isn't a valid react element. Source: https://facebook.github.io/react/docs/typechecking-with-proptypes.html

**Comment by yearofthedan** [comment]
On the face of it node is too lenient, in that a boolean - or text value for that matter - doesn't have props and style and cannot be laid out. 

But the example above highlights a common problem (I'm inferring) which is that you don't actually want to render the node, but React treats it as a valid child. So if we support non-element nodes being passed what would be the expected behaviour?  My first thought would be to allow all nodes, but filter non-elements from the array - this would also filter out text only nodes.

**Comment by yearofthedan** [comment]
This is implemented in 0.7.1. It now accepts, but filters out, any non-element nodes.

**Comment by camjackson** [comment]
Awesome!
